drivers/ieee802154/ieee802154_kw41z.c

```diff
@@ -313,6 +323,9 @@ static inline void kw41z_rx(struct kw41z_context *kw41z, u8_t len)
 
 	net_pkt_frag_insert(pkt, frag);
 
+#if CONFIG_SOC_MKW41Z4
+	u16_t reg_val = 0;
```

Comment:
looks like you could put this declaration at the beginning of the for loop. It's not super clean to declare a variable elsewhere than in the beginning of a code block